### DISCUSSION TITLE
I can't figure out how to use dynamic images :/

### DISCUSSION THREAD

**Discussion by kstraszewski** [nuxt/nuxt#18212] [Questions]
Hi im trying to use dynamic images in my nuxt3 beta project. Im trying good old require(`@/assets/${x}.webp`) but it seems that Vite doesn't understand require. So i found that solution:

```          
const imgUrl = new URL(`@/assets/banks/${props.offer.bank}-circle.webp`, import.meta.url)
```

this also doesn't work. So how can i make it work?

**UPDATE: (resolve but ugly):**

I found a solution here: https://github.com/vitejs/vite/issues/1265#issuecomment-885948466

I've added something like this to my setup function:
```

    const allImages = import.meta.globEager("../../assets/*.webp");
    const img = computed(() => {
        try {
          const x = allImages[`../../assets/${props.x}.webp`]
          return x.default;
        } catch (e) {
          console.warn(`Cannot find ${props.x}`, e)
        }
    })
```

**Comment by lukaszflorczak** [comment]
I'm also looking for the answer. The `const imgUrl = new URL(..., import.meta.url)` code is proper, but only for client-side rendering. There is also a note in Vite documentation:

> This pattern does not work if you are using Vite for Server-Side Rendering, because import.meta.url have different semantics in browsers vs. Node.js. The server bundle also cannot determine the client host URL ahead of time.

**Reply by alvarosabu** [reply]
You can use the [context](https://nuxtjs.org/docs/internals-glossary/context/) 

```
if (process.client) {
   imgUrl = new URL(..., import.meta.url)
 } else {
  // Something else, but couldn't figure it out yet
}
```
